src/aspire/utils/logging.py

```diff
@@ -81,5 +85,63 @@ def trange(*args, **kwargs):
     true/false will disable/enable tqdm progress bars.
     """
 
-    disable = config["logging"]["tqdm_disable"]
+    disable = config["logging"]["tqdm_disable"] or (
+        getConsoleLoggingLevel() not in ["DEBUG", "INFO"]
+    )
     return _tqdm.trange(*args, **kwargs, disable=disable)
+
+
+def setConsoleLoggingLevel(level_name):
+    """
+    Dynamically sets the console logging level by setting the level of the root logger's StreamHandler to `level_name`.
+    Note this will supersede the `logging.console_level` option stored in ASPIRE's configuration file.
+    :param level_name: One of "DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL".
+    """
+    if level_name not in LOGGING_LEVEL_NAMES:
+        raise ValueError(
+            f"{level_name} not a recognized logging level. Must be one of {LOGGING_LEVEL_NAMES}"
+        )
+    # handler list is ordered according to logging.conf
+    stream_handler = logging.getLogger().handlers[0]
+    stream_handler.setLevel(getattr(logging, level_name))
+
+
+def getConsoleLoggingLevel():
+    """
+    Returns the Python logging level of the root logger's StreamHandler, i.e. console output.
+    This is the same as the `logging.console_level` option in ASPIRE's configuration file unless
+    the console logging level has been changed dynamically during a session. (e.g. by a CLI option)
+    :return: The current console logging level name as a string. One of "DEBUG". "INFO", "WARNING",
```

Comment:
blank line between description and `return`

Reply:
thanks for catching, done